src/ansiblelint/rules/empty_string_compare.md

```diff
@@ -0,0 +1,44 @@
+# empty-string-compare
+
+This rule checks for empty string comparison in playbooks.
+To ensure code clarity you should avoid using empty strings in conditional statements with the ``when`` clause.
+
+* Use ``when: var|length > 0`` instead of ``when: var != ""``.
+* Use ``when: var|length == 0`` instead of ``when: var == ""``.
+
+This is an opt-in rule.
+You must enable it in your Ansible-lint configuration as follows:
+
+```yaml
+enable_list:
+  - empty-string-compare
+```
+
+## Problematic Code
+
+```yaml
+---
+- name: Example playbook
+  hosts: all
+  tasks:
+    - name: Shut down
+      ansible.builtin.command: /sbin/shutdown -t now
+      when: ansible_os_family == "" # <- Compares with an empty string.
+    - name: Shut down
+      ansible.builtin.command: /sbin/shutdown -t now
+      when: ansible_os_family !="" # <- Compares with an empty string.
+```
+
+## Correct Code
+
+```yaml
+---
+- name: Example playbook
+  hosts: all
+  tasks:
+    - name: Shut down
+      ansible.builtin.shell: |
+        /sbin/shutdown -t now
+        echo $var ==
+      when: ansible_os_family
```

Comment:
@ssbarnea I hope you don't read this on your PTO but I wanted to add a comment in this PR. I'm not sure what this is doing and it even seems to violate the rule. I asked about this in slack and have read through https://github.com/ansible/ansible-lint/issues/457 

IMO it would be good to clarify this example. Please let me know what you think (when you're back and ready to deal with it ofc). Thanks!